matlab/src/matlab/+arrow/+tabular/RecordBatch.m

```diff
@@ -54,13 +54,17 @@
         function arrowArray = column(obj, idx)
             import arrow.internal.validate.*
 
-            idx = index.numeric(idx, "int32");
-            % TODO: Consider vectorizing column() in the future to support
-            % extracting multiple columns at once.
-            validateattributes(idx, "int32", "scalar");
-
-            args = struct(Index=idx);
-            [proxyID, typeID] = obj.Proxy.getColumnByIndex(args);                
+            idx = index.numericOrString(idx, "int32");
+
+            if isnumeric(idx)
+                validateattributes(idx, "int32", "scalar");
```

Comment:
You can remove this `validateattributes` call if you include an `arguments` block:

```matlab
function arrowArray = column(obj, idx)
    arguments
         obj
         idx(1, 1)
    end
    ...
end
```
That way you don't need to re-validate the type of `idx`

Reply:
On second thought, we should enhance the `index.numericOrString` function to accept a `AcceptNonScalar` name-value pair that defaults to `true`.

Reply:
That's a good point. I've captured this in #37477.

Reply:
Marking this as resolved.